app/controllers/concerns/user_invitation.rb

```diff
@@ -62,6 +72,22 @@ def invite_new_user(email:, login: nil, first_name: nil, last_name: nil)
   end
 
   ##
+  # Sends a new invitation to the user with a new token.
+  #
+  # @param user_id [Integer] ID of the user to be re-invited.
+  # @return [Token] The new token used for the invitation.
+  def reinvite_user(user_id)
+    token = Token.find_by user_id: user_id, action: token_action
+
+    token.destroy! if token
+    new_token = Token.create! user_id: user_id, action: token_action
```

Comment:
But raising an internal error will leave the user clueless of the actual error in production mode. That can't be what you want?